test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,397 @@
+	host                 string
+}
+
+func (suite *TokenPropagationTestSuite) SetupSuite() {
+	t = suite.T()
+	if !isOpenShift(t) {
+		t.Skipf("Test %s is currently supported only on OpenShift because es-operator runs only on OpenShift\n", t.Name())
+	}
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &v1.JaegerList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+	}))
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &esv1.ElasticsearchList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Elasticsearch",
+			APIVersion: "logging.openshift.io/v1",
+		},
+	}))
+	var err error
+	ctx, err = prepare(t)
+	if err != nil {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+	addToFrameworkSchemeForSmokeTests(t)
+
+	suite.deployJaegerWithPropagationEnabled()
+
+}
+
+func (suite *TokenPropagationTestSuite) TearDownSuite() {
+	// undeployJaegerInstance(suite.exampleJaeger)
+	// handleSuiteTearDown()
+}
+
+func (suite *TokenPropagationTestSuite) TestTokenPropagationNoToken() {
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		req, err := http.NewRequest(http.MethodGet, suite.queryServiceEndPoint, nil)
+		resp, err := client.Do(req)
+		defer resp.Body.Close()
+		require.Equal(t, http.StatusForbidden, resp.StatusCode)
+		return true, nil
+	})
+	require.NoError(t, err, "Token propagation test failed")
+
+}
+
+func (suite *TokenPropagationTestSuite) TestTokenPropagationValidToken() {
+	/* Create an span */
+	collectorPort := randomPortNumber()
+	collectorPorts := []string{collectorPort + ":14268"}
+	portForwColl, closeChanColl := CreatePortForward(namespace, suite.collectorName, collectorPodImageName, collectorPorts, fw.KubeConfig)
+	defer portForwColl.Close()
+	defer close(closeChanColl)
+	collectorEndpoint := fmt.Sprintf("http://localhost:%s/api/traces", collectorPort)
+
+	cfg := config.Configuration{
+		Reporter:    &config.ReporterConfig{CollectorEndpoint: collectorEndpoint},
+		Sampler:     &config.SamplerConfig{Type: "const", Param: 1},
+		ServiceName: testServiceName,
+	}
+	tracer, closer, err := cfg.NewTracer()
+	require.NoError(t, err, "Failed to create tracer in token propagation test")
+
+	tStr := time.Now().Format(time.RFC3339Nano)
+	tracer.StartSpan("TokenTest").
+		SetTag("time-RFC3339Nano", tStr).
+		Finish()
+	closer.Close()
+
+	/* Get token using oauth OpenShift authorization */
+	client, err := oAuthAuthorization(suite.host, username, password)
+
+	/* Try to reach query endpoint */
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		req, err := http.NewRequest(http.MethodGet, suite.queryServiceEndPoint, nil)
```

Comment:
Same comment as before: make the test more resilient, otherwise it might become annoying to all future PRs ;)

Reply:
You probably want to check the `err` objects in this method and log their causes, without hard failing.